requestor/ext/requestor-gwtjackson/src/test/java/io/reinert/requestor/ext/gwtjackson/GwtJacksonProcessorTest.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2015 Danilo Reinert
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.reinert.requestor.ext.gwtjackson;
+
+import java.util.Arrays;
+
+import javax.tools.JavaFileObject;
+
+import com.google.testing.compile.JavaFileObjects;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static com.google.common.truth.Truth.assertAbout;
+import static com.google.testing.compile.JavaSourcesSubjectFactory.javaSources;
+
+@RunWith(JUnit4.class)
+public class GwtJacksonProcessorTest {
+
+    @Test
+    public void singleClass_CompilesSuccessfully() {
+        JavaFileObject pojoFile = JavaFileObjects.forSourceLines("test.Animal",
```

Comment:
At first I did think like that. But after studying a few projects that are doing annotation processing, I did find this way the best to unit test the processors. It's being applied in the dagger2 project this same way. Not moving the source lines in a a separate file promotes the benefit of readability. If tests happen to fail in the future, repairing them will be a lot easier if you have only one place to look at, instead of two.